test/integration/referral.spec.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ReferralError } from '../../src/ports/referral/types'
+import { test } from '../components'
+
+test('Referral', ({ components }) => {
+  const endpoint = 'https://referral-server.decentraland.zone/referral-progress'
+  let apiKey: string
+  let invitedUserAddress: string
+  let mockFetch: jest.Mock
+
+  beforeEach(async () => {
+    jest.clearAllMocks()
+
+    mockFetch = jest.fn()
+    global.fetch = mockFetch
+
+    apiKey = await components.config.requireString('REFERRAL_SERVER_API_KEY')
+    invitedUserAddress = '0x36359cd12e64c150a347f1dd3ff95bf68b46b33f'
+  })
+
+  describe('When updateReferral is called', () => {
+    describe('with valid body', () => {
+      beforeEach(() => {
+        mockFetch.mockResolvedValue({
+          ok: true,
+          status: 204
+        })
+      })
+
+      it('should update referral progress as signed up and return 204', async () => {
```

Comment:
As we're writting unit tests from the perspective of the component, we can't expect that it returns a 204, because that's the response of the API, not the `updateReferral` function.
You can only check for the request to have been completed successfully and the function to have resolved.
```suggestion
      it('should update referral progress as signed up and resolve', async () => {
```